src/Options/ParseOptions.hpp

```diff
@@ -647,6 +653,25 @@ std::string Parser<OptionList, Group>::help() const noexcept {
   return ss.str();
 }
 
+template <typename OptionList, typename Group>
+void Parser<OptionList, Group>::pup(PUP::er& p) noexcept {
+  static_assert(std::is_same_v<Group, NoSuchType>,
+                "Inner parsers should be recreated by the root parser, not "
+                "serialized.");
+  // We reconstruct most of the state when deserializing, rather than
+  // trying to package it.
+  p | help_text_;
+  p | input_source_;
+  if (p.isUnpacking() and not input_source_.empty()) {
+    auto received_source = std::move(input_source_);
+    input_source_.clear();
```

Comment:
For my understanding, why are you clearing the `input_source_`? doesn't this mean the next attempt to checkpoint-restart will fail to restore the parser? And why first moving to a temporary `received_source` and then moving each element from there to the parser?

Reply:
Under normal operation, `input_source_` is populated as files are parsed.  During unpacking we run the entire parsing sequence again, which will, as a (side?) effect, repopulate `input_source_`.

Reply:
Thanks for clarifying! I'd suggest adding a short comment like `// reparsing the input sources will repopulate input_source_, so clear it first` or similar, above this line.